common/parameters/Context.java

```diff
@@ -97,6 +100,15 @@ public Transaction id(long transactionId) {
         public long id() {
             return transactionId;
         }
+
+        public Transaction diagnosticTxn(ITransaction txn) {
```

Comment:
We attach the diagnostic txn so we can add things like query structure into from within the transaction later